packages/react-button/src/components/Button/Button.tsx

```diff
@@ -5,8 +5,35 @@ import { renderButton } from './renderButton';
 import { useButtonStyles } from './useButtonStyles';
 
 /**
- * Define a styled Button, using the `useButton` hook.
- * {@docCategory Button}
+ * Buttons give people a way to trigger an action.
+ *
+ * They’re typically found in forms, dialog panels, and dialogs. Some buttons are specialized for particular tasks,
```

Comment:
from experience authoring any kind of lengthy docs as JSDoc for storybook becomes hard to maintain.
- this applies for all inline "docs" in this story within this PR

The approach I'd recommend is to embrace markdown -> https://storybook.js.org/docs/react/writing-docs/docs-page#with-mdx-documentation

Also markdown is more universal format which can be used later for different docs approaches if we choose so.